Library/ViewModels/CommentsViewModel.swift

```diff
@@ -197,10 +197,20 @@ public final class CommentsViewModel: CommentsViewModelType,
       .ksr_debounce(.nanoseconds(0), on: AppEnvironment.current.scheduler)
 
     self.loadCommentsAndProjectIntoDataSource = Signal.merge(
-      // If we start off with no comments, show an empty state.
-      commentsAndProject.take(first: 1),
-      // Subsequent empty emissions (pull to refresh) should be ignored until replaced.
-      commentsAndProject.skip(first: 1).filter { comments, _ in comments.isEmpty == false }
+      // Allow empty arrays from the first emission.
+      Signal.zip(comments, initialProject)
+        .filter { $0.0.isEmpty }
+        .map { ($0.0, $0.1) }
```

Comment:
Lines 199-214 are doing a lot and it's a bit hard to follow along. I think it would be best to take the three separate signals you are merging here and put them into separate constants for readability.